src/core/shape_inference/include/embeddingbag_packed_shape_inference.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/validation_util.hpp"
+#include "openvino/op/util/embeddingbag_packed_base.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace util {
+
+template <class TShape>
+std::vector<TShape> shape_infer(const ov::op::util::EmbeddingBagPackedBase* op,
+                                const std::vector<TShape>& input_shapes) {
+    const auto input_size = input_shapes.size();
+    NODE_VALIDATION_CHECK(op, input_size == 2 || input_size == 3);
+
+    constexpr size_t EMB_TABLE = 0;
+    constexpr size_t INDICES = 1;
+    constexpr size_t PER_SAMPLE_WEIGHTS = 2;
+
+    auto indices_shape = input_shapes[INDICES];
+    NODE_VALIDATION_CHECK(op, indices_shape.rank().compatible(2), "INDICES must be 2D.");
+
+    if (input_size == 3) {
+        NODE_VALIDATION_CHECK(op,
+                              input_shapes[PER_SAMPLE_WEIGHTS].rank().compatible(2),
+                              "PER_SAMPLE_WEIGHTS must be 2D.");
+
+        NODE_VALIDATION_CHECK(op,
+                              TShape::merge_into(indices_shape, input_shapes[PER_SAMPLE_WEIGHTS]),
+                              "INDICES and PER_SAMPLE_WEIGHTS shape must be same.");
+    }
+    TShape output_shape;
+    const auto& emb_table_shape = input_shapes[EMB_TABLE];
+    if (emb_table_shape.rank().is_static()) {
+        NODE_VALIDATION_CHECK(op, emb_table_shape.size() > 0, "EMB_TABLE can't be a scalar.");
+        output_shape = emb_table_shape;
+        output_shape[0] = indices_shape.rank().is_static() ? indices_shape[0] : Dimension::dynamic();
+    } else {
```

Comment:
this else case can be removed if output shape set from emb shape.

Reply:
Updated approach is used in the common `copy_shape_and_update_dim` util.